core/src/main/scala/com/salesforce/op/stages/impl/insights/RecordInsightsLOCO.scala

```diff
@@ -161,7 +192,28 @@ class RecordInsightsLOCO[T <: Model[T]]
           val (indices, array) = aggregationMap.getOrElse(key, (Array.empty[Int], Array.empty[Double]))
           aggregationMap.update(key, (indices :+ i, sumArrays(array, diffToExamine)))
         }
-      } else {
+      }
+      // Let's check the descriptor value
+      // If those values exist, the field is likely to be a derived date feature from unit circle transformer.
+      else if (dateFeatureIndices.contains(oldInd) && history.descriptorValue.isDefined) {
+        // Name of the field
+        val rawName: Option[String] = history.parentFeatureType match {
```

Comment:
There is some code duplication. Can we avoid that?

Reply:
sure

Reply:
done.